leafsmart-front-end/src/components/buttons/AddFavourite.js

```diff
@@ -2,29 +2,65 @@ import { useAuth } from "../../contexts/AuthContext";
 import { useEffect, useState } from "react";
 
 const AddFavourite = (props) => {
-  const { geonameId, cName } = props
-  const [geoID, setGeoID] = useState(null)
-  const [displayName, setDisplayName] = useState("")
-  const [cityName, setCityName] = useState("")
+  const { geonameId, cName } = props;
+  const [geoID, setGeoID] = useState(null);
+  const [displayName, setDisplayName] = useState("");
+  const [cityName, setCityName] = useState("");
+  const [action, setAction] = useState("");
+  const [favs, setFavs] = useState([]);
 
-  const { saveFavourites } = useAuth();
+  const { saveFavourites, currentUser, returnFavourites, favourites } =
+    useAuth();
 
   useEffect(() => {
     setGeoID(geonameId);
     setDisplayName(cName);
     setCityName(cName);
   }, [geonameId, cName]);
 
+  useEffect(() => {
+    if (currentUser) {
+      returnFavourites();
+      setFavs(favourites);
+    }
+  }, [currentUser, returnFavourites, favourites]);
+
   const handleClick = (e) => {
     e.preventDefault();
-    saveFavourites(geoID, displayName, cityName)
-  }
+    saveFavourites(geoID, displayName, cityName);
+    setAction(!action);
+  };
+
+  const checkFavs = (cityName, fav) => {
+    const cityNameArr = [];
+    fav.forEach((item) => {
+      cityNameArr.push(item.city_name);
+    });
+    return cityNameArr;
+  };
+
+  const cityArr = checkFavs(cityName, favs);
 
   const { icon } = props;
   return (
-    <div className="w-16 h-16 bg-gradient-to-r from-Independence to-HeliotropeGray text-White flex items-center justify-center rounded-2xl hover:text-Isabelline hover:rounded-lg transition-all duration-300 ease-linear
-    cursor-pointer"><button onClick={handleClick}>{icon}</button></div>
+    <>
+      {cityArr.includes(cityName) ? (
+        <div
```

Comment:
This part could we pair/group tidy it a little? 
I'm wondering why the button is wrapped in a div -- could the classNames be applied directly to the button? I tried it as an experiment while I was adjusting the spacing on my other branch and the button continued to work.

Reply:
Adjusted to only use the button className